core/runtime/impl/wasm_memory_impl.cpp

```diff
@@ -42,22 +42,22 @@ namespace kagome::runtime {
     }
     if (new_offset <= memory_.size()) {
       offset_ = new_offset;
-      allocated[ptr] = size;
+      allocated_[ptr] = size;
       return ptr;
     }
 
     return freealloc(size);
   }
 
   std::optional<SizeType> WasmMemoryImpl::deallocate(WasmPointer ptr) {
-    const auto &it = allocated.find(ptr);
-    if (it == allocated.end()) {
+    const auto &it = allocated_.find(ptr);
+    if (it == allocated_.end()) {
       return std::nullopt;
```

Comment:
again, maybe use outcome::result?

Reply:
do you mean we should replace all std::optional with outcome result? Here we don't have too much reasons why function can fail, so optional is enough IMO